app/src/main/java/com/vero/tasky/agenda/presentation/agendaevent/EventDetailsViewModel.kt

```diff
@@ -168,11 +201,14 @@ class EventDetailsViewModel @Inject constructor(
                             remindAt = 0L
                         )
                         val agendaItem = state.agendaItem
+                        val attendees = agendaItem.attendees + newAttendee
                         updateState(
                             state.copy(
                                 agendaItem = agendaItem.copy(
-                                    attendees = agendaItem.attendees + newAttendee
+                                    attendees = attendees
                                 ),
+                                isGoingAttendees = attendees.filter { it.isGoing },
+                                isNotGoingAttendees = attendees.filter { !it.isGoing },
```

Comment:
You are running this filter in several places (here and on init)
You could have a "filterAttendees" or similar and run it there